libc/src/stdlib/memalignment.c

```diff
@@ -0,0 +1,19 @@
+#include "src/stdlib/memalignment.h"
+#include "src/__support/macros/config.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(size_t, memalignment, (const void *p)) {
+  if (p == NULL) {
+    return 0;
+  }
+
+  uintptr_t addr = (uintptr_t)p;
+
+  // Find the rightmost set bit, which represents the maximum alignment
+  // The alignment is a power of two, so we need to find the largest
+  // power of two that divides the address
+  return addr & (~addr + 1);
```

Comment:
```suggestion
  return addr & (~addr + 1);
```
I can't decide if it would be better to just use `ctz`. They reduce to the same ASM https://godbolt.org/z/qTjxc6dPn.

Reply:
Made up my mind, the bit twiddling is good here, while `1ull << cpp::count_trailing_zeroes(addr)` is probably easier to read, I think it performs worse for 64-bit values.

Reply:
then go with `__builtin_ctzg` is better?

Reply:
No, the opposite.

Reply:
does `1ull << cpp::count_trailing_zeroes(addr)` exist? I don't find it anywhere. what about using [`countr_zeros`](https://github.com/llvm/llvm-project/blob/main/libc/src/__support/CPP/bit.h#L81)?